src/main/utils/upload.ts

```diff
@@ -47,6 +48,96 @@ const getFileStream = async (filePath: string) => {
     return readableStream;
 };
 
+const getZipFileStream = async (
+    zip: StreamZip.StreamZipAsync,
+    filePath: string
+) => {
+    const stream = await zip.stream(filePath);
+    stream.pause();
+
+    let chunkToConsume = new Uint8Array(),
+        chunkArray: number[] = [],
+        closed = false;
+
+    let resolveObj: (value?: any) => void, rejectObj: (reason?: any) => void;
+
+    stream.on('data', (chunk: Uint8Array) => {
```

Comment:
Is there a `reader-based` approach can we use, instead of the `event-based` approach? in which we explicitly fetch the data rather than listening to an event

Reply:
I couldn't find any for this module.